pkg/api/validation/validation_test.go

```diff
@@ -1377,6 +1424,12 @@ func TestValidateService(t *testing.T) {
 	errs := ValidateService(&svc)
 	if len(errs) != 0 {
 		t.Errorf("Unexpected non-zero error list: %#v", errs)
+		for i := range errs {
+			t.Errorf("Found error: %s", errs[i].Error())
+		}
+	}
+	if svc.Spec.Protocol != "TCP" {
```

Comment:
defaulting should have all been removed - if there is any defaulting code left in validation, it' is broken